### PR TITLE
Add plugin e2e

### PR DESCRIPTION
**Pull request by alyssabull**
This PR adds the new `plugin-e2e` package that will eventually replace the `grafana/e2e`package (details in [this PR](https://github.com/grafana/grafana/pull/79969)). Upgrading the tests to the newest package will also address several issues associated with the `grafana/e2e` package that is currently not being maintained.

### CODE REVIEW

File: e2e/frontend/variableEditor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+//   variableEditPage.mockResourceResponse('api/v1/labels?*', prometheusLabels);
+//   await variableEditPage.datasource.set('gdev-prometheus');
+//   await variableEditPage.getByTestIdOrAriaLabel('Query type').fill('Label names');
+//   await page.keyboard.press('Tab');
+//   await variableEditPage.runQuery();
+//   await expect(
+//     variableEditPage,
+//     formatExpectError('Expected variable edit page to display certain label names after query execution')
+//   ).toDisplayPreviews(prometheusLabels.data);
```
Comment (bossinc) [inline e2e/frontend/variableEditor.spec.ts:25]:
```suggestion
```

**Review by bossinc — APPROVED** [review]
Nice! I like the documentation

File: e2e/frontend/configEditor.spec.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    // await page.getByTestId('data-testid Confirm Modal Danger Button').click();
+});
+
+  test('invalid auth token should throw valid error', async ({ createDataSourceConfigPage, page }) => {
+    const configPage = await createDataSourceConfigPage({ type: 'grafana-sentry-datasource' });
+    await page.getByPlaceholder('Sentry org slug').fill(SENTRY_ORG_SLUG);
+    await page.getByPlaceholder('Sentry Authentication Token').fill('invalid-auth-token');
+
+    await expect(configPage.saveAndTest()).not.toBeOK();
+    await expect(page.getByTestId('data-testid Alert error')).toHaveText('404 Not Found EOF');
```
Comment (aangelisc) [inline e2e/frontend/configEditor.spec.ts:35]: This should be a 401 based on the existing test

File: e2e/frontend/configEditor.spec.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const SENTRY_AUTH_TOKEN = 'mock-token';
+
+test.describe('Test config editor', () => {
+  test('empty configuration should throw valid error', async ({ createDataSourceConfigPage, page }) => {
+    const configPage = await createDataSourceConfigPage({ type: 'grafana-sentry-datasource' });
+
+    await expect(configPage.saveAndTest()).not.toBeOK();
+    await expect(page.getByTestId('data-testid Alert error')).toHaveText('invalid or empty organization slug');
+    
+    // await page.getByTestId('Data source settings page Delete button').click();
```
Comment (aangelisc) [inline e2e/frontend/configEditor.spec.ts:14]: I think it's fine to either include this commented code or remove it entirely, your call :)

File: e2e/frontend/variableEditor.spec.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { formatExpectError } from './errors';
+
+const SENTRY_E2E_PROJECT_NAME = "project-002";
+const SENTRY_E2E_PROJECT_ID = "proj002";
+
+test.describe('Sentry variables', () => {
+    test('add and edit variables', async ({ variableEditPage, page }) => {
+        await variableEditPage.getByTestIdOrAriaLabel('Select a data source').fill("sentry");
+        await page.keyboard.press('Enter');
+        // await variableEditPage.datasource.set('Sentry');
```
Comment (aangelisc) [inline e2e/frontend/variableEditor.spec.ts:11]: Is this intended to be commented?

**Pull request merged**